.github/workflows/deploy.yml

```
@@ -24,13 +24,16 @@ jobs:
   build-and-push:
     runs-on: ubuntu-latest
     timeout-minutes: 20
+    # Skip when packages/* workspaces are absent (e.g. waitlist-only forks).
+    # When packages/ is restored from upstream, the guard evaluates true and the job runs.
+    if: hashFiles('packages/shared/package.json') != ''
```

actionlint (1.7.12)

[error] calling function "hashFiles" is not allowed here. "hashFiles" is only available in "jobs.<job_id>.steps.continue-on-error", "jobs.<job_id>.steps.env", "jobs.<job_id>.steps.if", "jobs.<job_id>.steps.name", "jobs.<job_id>.steps.run", "jobs.<job_id>.steps.timeout-minutes", "jobs.<job_id>.steps.with", "jobs.<job_id>.steps.working-directory". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)